apps/src/sites/studio/pages/scripts/stage_extras.js

```diff
@@ -4,12 +4,36 @@ import StageExtras from '@cdo/apps/code-studio/components/stageExtras/StageExtra
 
 const script = document.querySelector('script[data-extras]');
 const config = JSON.parse(script.dataset.extras);
+const showProjectWidget = JSON.parse(script.dataset.widgetVisible);
+const projectTypes = JSON.parse(script.dataset.widgetTypes);
 
-ReactDOM.render(
-  <StageExtras
-    stageNumber={config.stageNumber}
-    nextLevelPath={config.nextLevelPath}
-    bonusLevels={config.bonusLevels}
-  />,
-  document.querySelector('#stage-extras')
-);
+// Get project data if showing the project widget.
+if (showProjectWidget) {
+  $.ajax({
+    method: 'GET',
+    url: `/v3/channels`,
+    dataType: 'json'
+  }).done(projectLists => {
+    ReactDOM.render(
+      <StageExtras
+        stageNumber={config.stageNumber}
+        nextLevelPath={config.nextLevelPath}
+        bonusLevels={config.bonusLevels}
+        showProjectWidget={showProjectWidget}
+        projectTypes={projectTypes}
+        projectList={projectLists}
+      />,
+      document.querySelector('#stage-extras')
```

Comment:
I wonder if this component should own retrieving it's own data, and displaying a loader gif while waiting for the AJAX request to complete?  Otherwise it becomes more difficult to re-use this component without copying over this initialization.